src/Modules/CrestApps.OrchardCore.AI.Chat/Hubs/AIChatHub.cs

```diff
@@ -118,6 +125,203 @@ await Clients.Caller.LoadSession(new
         });
     }
 
+    public async Task<string> SendAudioMessage(string profileId, string base64Audio, string sessionId, string sessionProfileId, CancellationToken cancellationToken)
+    {
+        try
+        {
+            if (string.IsNullOrWhiteSpace(profileId))
+            {
+                await Clients.Caller.ReceiveError(S["{0} is required.", nameof(profileId)].Value);
+                return null;
+            }
+
+            if (string.IsNullOrWhiteSpace(base64Audio))
+            {
+                await Clients.Caller.ReceiveError(S["Audio data is required."].Value);
+                return null;
+            }
+
+            // Limit audio size to prevent DoS attacks (only if limit is configured)
+            if (_chatOptions.MaxAudioSizeInBytes.HasValue && 
+                _chatOptions.MaxAudioSizeInBytes.Value > 0 && 
+                base64Audio.Length > _chatOptions.MaxAudioSizeInBytes.Value)
+            {
+                await Clients.Caller.ReceiveError(S["Audio data is too large. Please record a shorter message."].Value);
+                return null;
+            }
+
+            var profile = await _profileManager.FindByIdAsync(profileId);
+
+            if (profile is null)
+            {
+                await Clients.Caller.ReceiveError(S["Profile not found."].Value);
+                return null;
+            }
+
+            var httpContext = Context.GetHttpContext();
+
+            if (!await _authorizationService.AuthorizeAsync(httpContext.User, AIPermissions.QueryAnyAIProfile, profile))
+            {
+                await Clients.Caller.ReceiveError(S["You are not authorized to interact with the given profile."].Value);
+                return null;
+            }
+
+            // Get the speech-to-text client using the dedicated connection from profile metadata
+            var speechToTextMetadata = profile.As<CrestApps.OrchardCore.AI.Core.Models.SpeechToTextMetadata>();
```

Comment:
Use a using statement instead of using the full path. Apply this feedback everywhere